Extension/src/Debugger/configurationProvider.ts

```diff
@@ -392,32 +403,27 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
             });
         }));
         configs.push(defaultTemplateConfig);
-        // Get existing debug configurations from launch.json.
-        if (folder) {
-            const existingConfigs: vscode.DebugConfiguration[] | undefined = (await this.getLaunchConfigs(folder, type))?.map(config => ({
-                name: config.name,
-                type: config.type,
-                request: config.request,
-                detail: config.detail ? config.detail :
-                    config.preLaunchTask ? localize("pre.Launch.Task", "preLaunchTask: {0}", config.preLaunchTask) : undefined,
-                existing: TaskConfigStatus.configured,
-                preLaunchTask: config.preLaunchTask
-            }));
-            if (existingConfigs) {
-                const areEqual = (config1: vscode.DebugConfiguration, config2: vscode.DebugConfiguration): boolean =>
-                    (config1.preLaunchTask === config2.preLaunchTask
-                    && config1.type === config2.type && config1.request === config2.request);
-                // Remove the detected configs that are already configured once in launch.json.
-                const dedupExistingConfigs: vscode.DebugConfiguration[] = configs.filter(detectedConfig => !existingConfigs.some(config => {
-                    if (areEqual(config, detectedConfig)) {
-                        // Carry the default task information.
-                        config.isDefault = detectedConfig.isDefault ? detectedConfig.isDefault : undefined;
-                        return true;
-                    }
-                    return false;
-                }));
-                configs = existingConfigs.concat(dedupExistingConfigs);
+        const existingConfigs: vscode.DebugConfiguration[] | undefined = (await this.getLaunchConfigs(folder, type))?.map(config => {
+            if (!config.detail && config.preLaunchTask) {
+                config.detail = localize("pre.Launch.Task", "preLaunchTask: {0}", config.preLaunchTask);
             }
+            config.existing = TaskConfigStatus.configured;
+            return config;
+        });
+        if (existingConfigs) {
+            const areEqual = (config1: vscode.DebugConfiguration, config2: vscode.DebugConfiguration): boolean =>
+                (config1.preLaunchTask === config2.preLaunchTask
+                && config1.type === config2.type && config1.request === config2.request);
+            // Remove the detected configs that are already configured once in launch.json.
+            const dedupExistingConfigs: vscode.DebugConfiguration[] = configs.filter(detectedConfig => !existingConfigs.some(config => {
+                if (areEqual(config, detectedConfig)) {
```

Comment:
Why is areEqual used when it could just directly check config properties? Seems like that just adds more code, unless areEqual was used somewhere else.